components/admin_console/license_settings/modals/upload_license_modal.tsx

```diff
@@ -0,0 +1,281 @@
+        e.preventDefault();
+        if (fileObj === null) {
+            return;
+        }
+
+        setIsUploading(true);
+        const {error} = await dispatch(uploadLicense(fileObj));
+
+        if (error) {
+            setFileObj(null);
+            setServerError(error.message);
+            setIsUploading(false);
+            return;
+        }
+
+        await dispatch(getLicenseConfig());
+        setFileObj(null);
+        setServerError(null);
+        setIsUploading(false);
+    };
+
+    const show = useSelector((state: GlobalState) => isModalOpen(state, ModalIdentifiers.UPLOAD_LICENSE));
+    if (!show) {
+        return null;
+    }
+
+    const handleOnClose = () => {
+        if (props.onExited) {
+            props.onExited();
+        }
+        dispatch(closeModal(ModalIdentifiers.UPLOAD_LICENSE));
+    };
+
+    const handleRemoveFile = () => {
+        setFileObj(null);
+    };
+
+    let uploadLicenseContent = (
+        <>
+            <div className='content-body'>
+                <div className='svg-image'>
+                    <WomanArmOnTable
+                        width={200}
+                        height={200}
+                    />
+                </div>
+                <div className='title'>
+                    <FormattedMessage
+                        id='admin.license.upload-modal.title'
+                        defaultMessage='Upload a License Key'
+                    />
+                </div>
+                <div className='subtitle'>
+                    <FormattedMessage
+                        id='admin.license.upload-modal.subtitle'
+                        defaultMessage='Upload a license key for Mattermost Enterprise Edition to upgrade this server. '
+                    />
+                </div>
+                <div className='file-upload'>
+                    <div className='file-upload__titleSection'>
+                        <FormattedMessage
+                            id='admin.license.upload-modal.file'
+                            defaultMessage='File'
+                        />
+                    </div>
+                    <div className='file-upload__inputSection'>
+                        <div className='help-text file-name-section'>
+                            {fileObj?.name && fileObj?.size ? (
+                                <>
+                                    <FileSvg
+                                        width={20}
+                                        height={20}
+                                    />
+                                    <span className='file-name'>
+                                        {fileObj.name.length < 40 ? fileObj?.name : `${fileObj?.name.substr(0, 37)}...`}
+                                    </span>
+                                    <span className='file-size'>
+                                        {(fileObj?.size / 1024).toFixed(2) + 'MB'}
```

Comment:
Can we reuse `fileSizeToString` from utils/utils.jsx here? It is more robust. It rounds decimals instead of showing thhem, but it will handle files that are KB in size and display them as KB.

Reply:
sure!